x/protorev/keeper/epoch_hook.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	epochstypes "github.com/osmosis-labs/osmosis/v13/x/epochs/types"
+	"github.com/osmosis-labs/osmosis/v13/x/protorev/types"
+)
+
+type EpochHooks struct {
+	k Keeper
+}
+
+// Struct used to track the pool with the highest liquidity
+type LiquidityPoolStruct struct {
+	Liquidity sdk.Int
+	PoolId    uint64
+}
+
+var (
+	_ epochstypes.EpochHooks = EpochHooks{}
+)
+
+func (k Keeper) EpochHooks() epochstypes.EpochHooks {
+	return EpochHooks{k}
+}
+
+// BeforeEpochStart is the epoch start hook.
+func (h EpochHooks) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	return nil
+}
+
+// AfterEpochEnd is the epoch end hook. The module will update all of the pools in the store that are
+// used for trading.
+func (h EpochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	switch epochIdentifier {
```

Comment:
This switch statement will get populated with more info as PRs are submitted.

Reply:
In the current form, it's just recomputing and storing the highest liquidity pool for each token x {osmo, atom}. 

This seems like most of the logic that would happen on an epoch to me. @davidterpay what's missing?

Reply:
Given that we wouldn't be able to see a swap on a deprecated pool, the route building logic we have can potentially build a route with a stale/deprecated pool which is why we want to be able to delete all of the deprecated pools from the store before resetting it in state. 

In terms of the additional functionality for the epoch hook, it will also be populated with functionality that distributes a portion of the developer fees to a developer account (which presumably will be skip's address once deployed). This requires us to track various info such as the number of days since genesis so that we can properly split the arbitrage revenue generated at any given moment in time.